velocity.js

```diff
@@ -525,29 +525,70 @@
 			return result;
 		}
 
+		/**
+		 * Shim for "fixing" IE's lack of support (IE < 9) for applying slice
+		 * on host objects like NamedNodeMap, NodeList, and HTMLCollection
+		 * (technically, since host objects have been implementation-dependent,
+		 * at least before ES2015, IE hasn't needed to work this way).
+		 * Also works on strings, fixes IE < 9 to allow an explicit undefined
+		 * for the 2nd argument (as in Firefox), and prevents errors when
+		 * called on other DOM objects.
+		 */
 		var _slice = (function() {
-			var slice = Array.prototype.slice;
+			var _fslice = Array.prototype.slice;
 
 			try {
 				// Can't be used with DOM elements in IE < 9
-				slice.call(document.documentElement);
+				_fslice.call(document.documentElement);
 			} catch (e) { // Fails in IE < 9
 				// This will work for genuine arrays, array-like objects, 
 				// NamedNodeMap (attributes, entities, notations),
 				// NodeList (e.g., getElementsByTagName), HTMLCollection (e.g., childNodes),
 				// and will not fail on other DOM objects (as do DOM elements in IE < 9)
-				slice = function() {
-					var i = this.length,
-							clone = [];
+				Array.prototype.slice = function(begin, end) {
```

Comment:
I don't agree. Slice is supported in *modern* browsers, but that excludes IE8 (a relic that we still support for the next couple of years). And thus users using slice on IE8 will be using a similar shim to gain that ability on IE8. Modern browsers won't have their implementation of slice replaced at all (because the catch() block won't be called).